src/validationinterface.h

```diff
@@ -189,16 +155,53 @@ class CValidationInterface {
     friend class ValidationInterfaceTest;
 };
 
-class MainSignalsImpl;
+/**
+ * MainSignalsImpl manages a list of shared_ptr<CValidationInterface> callbacks.
```

Comment:
Felt more natural to give a `CMainSignals` ownership of `MainSignalsImpl` instead of a `std::unique_ptr<MainSignalsImpl`, so it needs the full declaration of the class.

Reply:
Ah, I think we need the full declaration here, because other modules are now instantiating the `CMainSignals`.